### PR TITLE
Bump MariaDB versions

### LINKED ISSUE #55: Bump all MariaDB versions to latest
To fix [incompatibilities with dump files](https://mariadb.org/mariadb-dump-file-compatibility-change/), we should upgrade all (supported) MariaDB versions to latest. It seems that Bitnami is offering newer MariaDB versions only in `debian12` flavour, but this shouldn't be a problem?

This is a problem when generating a dump in production, running latest LTS version of MariaDB, and trying to import it in Lando.

```
Importing /app/db.sql...
ERROR at line 1: Unknown command '\-'.
Import failed.
```

Sure we can fix this running `sed -i '1{/999999.*sandbox/d}' db.sql` locally before running `/helpers/sql-import.sh`, but preferably this should work out of the box.

### PR DESCRIPTION
**Pull request by todeveni**
Fixes #55 

### Bare minimum self-checks

- [x] I've updated this PR with the latest code from `main`
- [x] I've done a cursory QA pass of my code locally
- [x] I've ensured all automated status check and tests pass
- [x] I've [connected this PR to an issue](https://help.zenhub.com/support/solutions/articles/43000010350-connecting-pull-requests-to-github-issues)

### Pieces of flare

- [ ] I've written a unit or functional test for my code
- [ ] I've updated relevant documentation it my code changes it
- [ ] I've updated this repo's README if my code changes it
- [x] I've updated this repo's CHANGELOG with my change unless its a trivial change (like updating a typo in the docs)

### Finally

- [ ] I've [requested a review](https://help.github.com/en/articles/requesting-a-pull-request-review) with relevant people

If you have any issues or need help please join the `#contributors` channel in the [Lando slack](https://www.launchpass.com/devwithlando) and someone will gladly help you out!

You can also check out the [coder guide](https://docs.lando.dev/contrib/coder.html).

### CODE REVIEW

File: builders/mariadb.js [hunk 1]
```diff
@@ -11,22 +11,22 @@ module.exports = {
     version: '10.3',
     supported: ['11.3', '11.2', '11.1', '11.0', '10.11', '10.10', '10.9',
       '10.8', '10.7', '10.6', '10.5', '10.4', '10.3', '10.2', '10.1'],
-    legacy: ['10.10', '10.9', '10.8', '10.7', '10.3', '10.2', '10.1'],
+    legacy: ['11.3', '10.10', '10.9', '10.8', '10.7', '10.3', '10.2', '10.1'],
```
Comment (uberhacker) [inline builders/mariadb.js:14]: Wouldn't legacy be 11.0, 11.1 and 11.2?

Comment (todeveni) [reply builders/mariadb.js:14]: Oh, that isn't supposed to be there. 11.0 will become EOL in a few weeks.

I looked at https://endoflife.date/mariadb listing, which incorrectly states 11.3 as EOL. It's a rolling release (with no set EOL date?).

File: examples/10.3/README.md [hunk 2]
```diff
@@ -22,8 +22,8 @@ Verification commands
 Run the following commands to validate things are rolling as they should.
 
 ```bash
-# Should use 10.3.27 as the default version
-lando ssh -s defaults -c "mysql -V | grep 10.3.27"
+# Should use 10.3.39 as the default version
+lando ssh -s defaults -c "mysql -V | grep 10.3.39"
```
Comment (uberhacker) [inline examples/10.3/README.md:26]: Change `mysql` to `mariadb`?

File: examples/10.2/README.md [hunk 3]
```diff
@@ -22,8 +22,8 @@ Verification commands
 Run the following commands to validate things are rolling as they should.
 
 ```bash
-# Should use 10.2.36 as the default version
-lando ssh -s defaults -c "mysql -V | grep 10.2.36"
+# Should use 10.2.44 as the default version
+lando ssh -s defaults -c "mysql -V | grep 10.2.44"
```
Comment (uberhacker) [inline examples/10.2/README.md:26]: Change `mysql` to `mariadb`?

File: examples/10.3/README.md (hunk 2 above)
Comment (todeveni) [reply examples/10.3/README.md:26]: `mariadb` binaries/symlinks are [available only on 10.4+](https://jira.mariadb.org/browse/MDEV-21303)

**Review by uberhacker — APPROVED** [review]
The important thing to note here is the mariadb command was introduced in 10.4 so any earlier versions must use the mysql command.

**Comment by stephen-cox** [comment]
It's great to see a fix for #55 - is there a timeline for a new release which include this fix?

**Comment by reynoldsalec** [comment]
Just released a new MariaDB with a few other fixes @stephen-cox. https://github.com/lando/mariadb/releases/tag/v1.4.0

Going to need to update that service across recipes as well.

**Pull request merged**